### DISCUSSION TITLE
[Contest Logging] Restoring a Session

### DISCUSSION THREAD

**Discussion by magicbug** [magicbug/Cloudlog#776] [Ideas / Feature Requests]
At the moment with contest logging, if you close the session you lose the ability to go back which is a faff so we need to figure out the best way to store a session.

I'm thinking the easiest way is to use localStorage in Javascript, we can store the first QSO of the session (date/time/callsign/contest-id) and also the options for the exchange.

On page load, we get the local storage variables and pass them to MySQL to get the QSO list, and use the rest to build the exchange information.

@AndreasK79 thoughts?

**Comment by AndreasK79** [comment]
Yep. That is a possibility. I was thinking PHP session, or a DB table, but I guess localStorage is probably even easier.

**Comment by AndreasK79** [comment]
That was easy to use, now it's just a matter of writing the logic.

When should we store this? 

- After first QSO is logged?
- After setting of exchange type?
- After setting of contest?
- All of the above?

Restoring is just a matter of calling the function on document.ready() which then sets the form and fills the table with data.

**Reply by magicbug** [reply]
I'd suggest setting things at each stage, so "on change" of Contest ID and Exchange Type then store QSO info on first save then also update the localstrage for the exchange info so the maths for serial would be done within the frontend

**Reply by AndreasK79** [reply]
Exchange sent, exchange type and contest name is done. Now I need to work out the logic for the log table.

**Reply by AndreasK79** [reply]
I added a PR now with exchange sent, exchange type and contest name.

As for the QSOs, I think we need to decide how we should do this:

- When should we delete the contest session?
- Should all QSOs be loaded? This might be hundreds or thousands of QSOs. 
- If all QSOs is loaded, should we add datatable for a scrollable and searchable table?
- Should a we add a "Reset contest session" button, or should we have a timeout value?

**Reply by magicbug** [reply]
I'd suggest we just have a reset button once users done they can just clear it.

Data table for QSOs seems a good idea, I'd think there wouldn't be 1000s probably low hundreds unless someone really loves the basic features 🤷‍♂️

**Reply by AndreasK79** [reply]
You are right. If people are logging that many QSOs, they would be better of using a contest program.

Things are coming along. Button is added and table is filled. Just a few tings to work out. When that is done, I'll look into adding datatable for the table.
